From d7e756df08dd200a01222aef81866dab8d6087e9 Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@canonical.com>
Date: Wed, 19 Dec 2007 10:31:25 +0000
Subject: [PATCH] [EVAL] Fix unaligned trap on sparc

dash dies on sparc with a SIGBUS due to an arithmetic error introduced
with commit 03b4958, this patch fixes it.
---
 src/eval.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/eval.c b/src/eval.c
index a8feaa0..fd2c394 100644
--- a/src/eval.c
+++ b/src/eval.c
@@ -722,7 +722,7 @@ evalcommand(union node *cmd, int flags)
 	}
 
 	/* Reserve one extra spot at the front for shellexec. */
-	argv = nargv = stalloc(sizeof (char *) * (argc + 2)) + 1;
+	argv = nargv = ((char **)stalloc(sizeof (char *) * (argc + 2))) + 1;
 	for (sp = arglist.list ; sp ; sp = sp->next) {
 		TRACE(("evalcommand arg: %s\n", sp->text));
 		*nargv++ = sp->text;
-- 
1.5.3.7

